### DISCUSSION TITLE
Which addresses / domain do I need to whitelist for Immich to download machine learning models?

### DISCUSSION THREAD

**Discussion by SayantanRC** [immich-app/immich#14068] [General]
I have a script that blocks all network access except for a few allowed domains. Consider this as a sort of firewall. I Whitelisted 2 domains / addresses for the machine learning docker container

1. huggingface.co
2. hf-mirror.com

However, this was insufficient as the docker logs showed that the container failed to download the models after about 33%. What are the other addresses I should whitelist?

**Comment by mertalev** [comment]
Hugging Face is the only thing we make requests to in order to download models. It's possible that some requests go to a CDN etc. with a different domain.

**Reply by SayantanRC** [reply]
Would have been excellent if we could get to know that domain.

**Reply by mertalev** [reply]
It isn't something we determine, so this isn't really the right place to ask about that. An alternative is to let the container download the model without restrictions, then block internet access altogether. It doesn't need internet access after the model is downloaded.

**Reply by SayantanRC** [reply]
How do I download the model beforehand? From the docker logs, I see it only downloads the models when I first start an ML related job like face detection.

**Reply by mertalev** [reply]
You can upload any random image and it'll download the models in order to process it.
